### PR TITLE
Fix how buttons/links are decorated

### PR DESCRIPTION
**Pull request by amglebihan**
Test URLs:
- Before: https://main--vg-volvotrucks-us--hlxsites.aem.page/
- After: https://fix-buttons-decoration--vg-volvotrucks-us--hlxsites.aem.page/

### CODE REVIEW

**Review by jonatan-lledo-netcentric — APPROVED** [review]

**Review by santi-homps — DISMISSED** [review]

**Review by jonatan-lledo-netcentric — APPROVED** [review]

File: scripts/scripts.js [hunk 1]
```diff
@@ -656,11 +656,19 @@ const addClassToContainer = (element) => {
...
     if (shouldDecorateLink(a)) {
-      const up = a.parentElement;
+      let up = a.parentElement;
       const twoUp = up.parentElement;
       const threeUp = twoUp.parentElement;
       const buttonClass = getButtonClass(up, twoUp);
-      a.className = `button ${buttonClass}`;
+
+      // Redefine the up element after calling reparentChildren
+      up = a.parentElement;
```
Comment (alexiscoelho) [inline scripts/scripts.js:665]: Why this?

Comment (amglebihan) [reply scripts/scripts.js:665]: The "reparentChildren" function, which is called in "getButtonClass" above, removes unnecessary "em" and "strong" tags around anchors. As a result, we need to redefine "up" after these tags are removed.

Comment (alexiscoelho) [reply scripts/scripts.js:665]: Wow, that getButtonClass method is really misleading :/

**Comment by amglebihan** [comment]
@cogniSyb Regarding your comment above, I added a condition that checks if the parent elements of an anchor doesn't have the "button-container" class and remove the button classes if it is the case.

**Review by jonatan-lledo-netcentric — APPROVED** [review]

**Pull request merged**